flink-1.17-connector-bigquery/flink-connector-bigquery/src/test/java/com/google/cloud/flink/bigquery/table/BigQueryDynamicTableSinkTest.java

```diff
@@ -0,0 +1,157 @@
+        BigQueryDynamicTableSink bigQueryDynamicTableSink =
+                new BigQueryDynamicTableSink(bigQuerySinkConfig, logicalTypeSchema);
+        BigQueryDynamicTableSink bigQueryDynamicTableSinkCopy =
+                new BigQueryDynamicTableSink(bigQuerySinkConfig, logicalTypeSchema);
+
+        assertTrue(checkEquality(bigQueryDynamicTableSinkCopy, bigQueryDynamicTableSink.copy()));
+    }
+
+    @Test
+    public void testSummaryString() {
+        BigQueryDynamicTableSink bigQueryDynamicTableSink =
+                new BigQueryDynamicTableSink(bigQuerySinkConfig, logicalTypeSchema);
+        assertEquals("BigQuery", bigQueryDynamicTableSink.asSummaryString());
+    }
+
+    @Test
+    public void testChangelogMode() {
+        BigQueryDynamicTableSink bigQueryDynamicTableSink =
+                new BigQueryDynamicTableSink(bigQuerySinkConfig, logicalTypeSchema);
+        assertEquals(
+                ChangelogMode.insertOnly(),
+                bigQueryDynamicTableSink.getChangelogMode(Mockito.mock(ChangelogMode.class)));
+    }
+
+    @Test
+    public void testSinkRuntimeProvider() {
+        BigQueryDynamicTableSink bigQueryDynamicTableSink =
+                new BigQueryDynamicTableSink(bigQuerySinkConfig, logicalTypeSchema);
+        assertInstanceOf(
+                SinkV2Provider.class,
+                bigQueryDynamicTableSink.getSinkRuntimeProvider(
+                        Mockito.mock(DynamicTableSink.Context.class)));
+    }
+
+    /**
+     * Method overwritten to check equality, required for testing.
+     *
+     * @param bigQueryDynamicSink1 Object 1.
+     * @param bigQueryDynamicSink2 Target Object to check equality with.
+     * @return True if {@link Object} is equal to current object.
+     */
+    private boolean checkEquality(Object bigQueryDynamicSink1, Object bigQueryDynamicSink2) {
+        if (bigQueryDynamicSink1 == bigQueryDynamicSink2) {
+            return true;
+        }
+        if (bigQueryDynamicSink1.getClass() != bigQueryDynamicSink2.getClass()) {
+            return false;
+        }
+        BigQueryDynamicTableSink bigQueryDynamicTableSink1 =
+                (BigQueryDynamicTableSink) bigQueryDynamicSink1;
+        BigQueryDynamicTableSink bigQueryDynamicTableSink2 =
+                (BigQueryDynamicTableSink) bigQueryDynamicSink2;
+        BigQuerySinkConfig sinkConfig1 = bigQueryDynamicTableSink1.getSinkConfig();
+        BigQuerySinkConfig sinkConfig2 = bigQueryDynamicTableSink2.getSinkConfig();
+
+        if ((bigQueryDynamicTableSink1.getLogicalType()
+                        == bigQueryDynamicTableSink2.getLogicalType())
+                && (sinkConfig1.getConnectOptions() == sinkConfig2.getConnectOptions())
+                && (sinkConfig1.getSerializer() == sinkConfig2.getSerializer())
+                && (sinkConfig1.getDeliveryGuarantee() == sinkConfig2.getDeliveryGuarantee())) {
+            BigQuerySchemaProvider thisSchemaProvider = sinkConfig1.getSchemaProvider();
```

Comment:
Be consistent with the variable names. Instead of 'thisSchemaProvider' just use 'schemaProvider1'. As this is not an instance method, no need to use names that start with 'this'.

Reply:
ACK, moved to the class.